[DOCS] Adds settings of question_answering to inference_config of PUT and infer trained model APIs (#86895)
elastic/elasticsearch: docs/reference/ml/trained-models/apis/put-trained-models.asciidoc

======
=====

`question_answering`:::
(Optional, object)
include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-question-answering]
+
.Properties of question_answering inference
[%collapsible%open]
=====
`max_answer_length`::::
(Optional, integer)

> The maximum amount of words that the answer consists. Defaults to `15`.

Reviewer: Might be simpler as:
```suggestion
The maximum amount of words in the answer. Defaults to `15`.
```

`tokenization`::::
(Optional, object)
include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-tokenization]

> Recommended to set `max_sentence_length` to `386` with `128` of `span` and set 

Reviewer: This is not indented properly at the moment, but it might be as simple as adding a continuation character:
```suggestion
+
Recommended to set `max_sentence_length` to `386` with `128` of `span` and set 
```

`truncate` to `none`.

> +

Reviewer: > +

This is currently not displaying properly. If my other suggestion doesn't solve this problem, we might need to put this whole tokenization description within a "--" block. Lmk if you want me to help play

======
=====

`question_answering`:::
(Optional, object)
include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=inference-config-nlp-question-answering]
+
.Properties of question_answering inference
[%collapsible%open]
=====
`max_answer_length`::::
(Optional, integer)
The maximum amount of words that the answer consists. Defaults to `15`.

`question`::::
(Required, string)
The question that you want to be answered based on the context of the provided 

> text.

Reviewer 2: `question` is not settable on PUT. This makes question_answering unique. `question` is only settable at inference time.
